### PR TITLE
[WIP] Symfony 3.0 upgrade 

### PR DESCRIPTION
**Pull request by umpirsky**
I am testing [Symfony Upgrade Fixer](https://github.com/umpirsky/Symfony-Upgrade-Fixer) against several Symfony projects.

Will continue pushing commits to this PR, I hope you will find them useful.

| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | yes |
| Deprecations? | no |
| License | MIT |

### CODE REVIEW

**Comment by stloyd** [comment]
Problem with some changes is that it will not work for Symfony `2.7` which is LTS, so supporting `3.0` is good idea but in fact it's using partially `2.8` which is not out there yet.

**Comment by umpirsky** [comment]
@stloyd Yes, I know, that's why I tagged it with `Future` tag. :)

File: src/Sylius/Bundle/ReportBundle/Form/Type/Renderer/TableConfigurationType.php [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 namespace Sylius\Bundle\ReportBundle\Form\Type\Renderer;
 
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ReportBundle/Form/Type/Renderer/TableConfigurationType.php]: @umpirsky Would be cool if this is sorted alphabetically. :)

Comment (umpirsky) [inline src/Sylius/Bundle/ReportBundle/Form/Type/Renderer/TableConfigurationType.php]: @pjedrzejewski https://github.com/umpirsky/Symfony-Upgrade-Fixer/issues/new :P

Comment (pjedrzejewski) [inline src/Sylius/Bundle/ReportBundle/Form/Type/Renderer/TableConfigurationType.php]: Hehe, done - https://github.com/umpirsky/Symfony-Upgrade-Fixer/issues/13 :)

Comment (umpirsky) [inline src/Sylius/Bundle/ReportBundle/Form/Type/Renderer/TableConfigurationType.php]: @pjedrzejewski Thanks! :)

**Comment by gperdomor** [comment]
Here is a list of symfony bundles working on symfony 3 https://docs.google.com/spreadsheets/d/1GOxJ2lNpLxN12YyZ87Bhsxb0iLAQ7e4UDZbJSly8Sy8/edit?pli=1#gid=0

**Comment by webda2l** [comment]
About A2lixTranslationFormBundle, if you want continue with it, it's in progress (https://github.com/a2lix/TranslationFormBundle/pull/205) and will need to change it require version to ~3.0 when released

**Comment by bendavies** [comment]
Why don't you first target 2.8 *and* fix the deprecations?

**Comment by umpirsky** [comment]
@bendavies Good point, maybe I can split after Symfony-Upgrade-Fixer supports [specifying target version](https://github.com/umpirsky/Symfony-Upgrade-Fixer/issues/3).

**Comment by adamelso** [comment]
Another idea could be to split the PR to upgrade each package, starting with the Resource component and bundle.

**Comment by rvanlaak** [comment]
Any news on this one? Would be awesome if there could be some more info on the planning on this one?

**Comment by pamil** [comment]
@rvanlaak right now there are two main issues:
 - using forms as services in ResourceBundle - it may be solved by https://github.com/isometriks/LegacyFormBundle
 - our new Behat setup uses dependency injection scopes which was deleted in Symfony 3.0

**Comment by rvanlaak** [comment]
At a certain moment you'll need to get rid of the aliases and scopes right? So wouldn't it be better to start on removing and fixing them right away? 

I didn't follow the decision process on that, but I'd advise to not implement the two fixes you propose. What do you think?

**Comment by rvanlaak** [comment]
@pjedrzejewski you have news on https://github.com/Sylius/Sylius/pull/3571#issuecomment-224835803 ?

**Comment by rvanlaak** [comment]
@pamil was the Behat dependency on scopes resolved in the meanwhile? 👍

**Comment by pamil** [comment]
@rvanlaak I'm in progress of making a PoC :)

**Comment by michalmarcinkowski** [comment]
Closing in favor of https://github.com/Sylius/Sylius/pull/6738.

**Pull request closed without merging**